### PR TITLE
Fix build on ubuntu

### PR DESCRIPTION
**Pull request by wangshilong**
Mainly cherry-pick from mainstream, Fixed that DAOS lib could not be found when
installing from source codes.

Signed-off-by: Wang Shilong <shilong.wang@intel.com>

### CODE REVIEW

**Comment by wangshilong** [comment]
@mchaarawi would you merge this to your branch?

File: cmake/FindCART.cmake [hunk 1]
```diff
@@ -0,0 +1,38 @@
+# - Try to find cart libs
```
Comment (mchaarawi) [inline cmake/FindCART.cmake:1]: @wangshilong why do we need a separate find_cart here?
cart is installed with DAOS, so can it be just integrated with find_daos like the other libraries (dfs, etc)?

Comment (wangshilong) [reply cmake/FindCART.cmake:1]: @mchaarawi I think no from source installing, but this is there in main stream, i guessed story is a bit different from RPM installing? so i picked it here too.

Comment (mchaarawi) [reply cmake/FindCART.cmake:1]: i think both source and RPM install put daos and cart together.
i guess this needs fixing in mfu mainstream, ill check with danielle or dalton

Comment (mchaarawi) [reply cmake/FindCART.cmake:1]: actually seems like this is not needed anymore.
https://github.com/hpc/mpifileutils/commit/bf4f89631d1a2a276e3cdbec5b141969c3f6f5a8
removed the with cart option, but just missed deleting this fail. so this file is not used i would think

**Pull request merged**